src/classes/showExpensesListForContactController.cls

```diff
@@ -0,0 +1,39 @@
+public class showExpensesListForContactController {
+    
+    private String queryString = '';
+    
+    public List<Expense__c> getExpenses(){
+        
+        
+        String idContact = ApexPages.currentPage().getParameters().get('id');
```

Comment:
@pacosta1987 I don't see the changes I requested being done here. For instance, the queries still seem to be unlimited, and no pagination has been implemented.